domains/pallets/ethereum-beacon-client/src/lib.rs

```diff
@@ -32,36 +32,46 @@ use sp_std::prelude::*;
 
 pub use pallet::*;
 
-pub type BlockUpdateOf<T> = BlockUpdate<
-    <T as Config>::MaxFeeRecipientSize,
-    <T as Config>::MaxLogsBloomSize,
-    <T as Config>::MaxExtraDataSize,
-    <T as Config>::MaxDepositDataSize,
-    <T as Config>::MaxPublicKeySize,
-    <T as Config>::MaxSignatureSize,
-    <T as Config>::MaxProofBranchSize,
-    <T as Config>::MaxProposerSlashingSize,
-    <T as Config>::MaxAttesterSlashingSize,
-    <T as Config>::MaxVoluntaryExitSize,
-    <T as Config>::MaxAttestationSize,
-    <T as Config>::MaxValidatorsPerCommittee,
-    <T as Config>::MaxSyncCommitteeSize,
+pub type BlockUpdateOf = BlockUpdate<
+    config::FeeRecipientSize,
+    config::BytesPerLogsBloom,
+    config::MaxExtraDataBytes,
+    config::MaxDeposits,
+    config::PublicKeySize,
+    config::SignatureSize,
+    config::DepositContractTreeDepth,
+    config::MaxProposerSlashings,
+    config::MaxAttesterSlashings,
+    config::MaxVoluntaryExits,
+    config::MaxAttestations,
+    config::MaxValidatorsPerCommittee,
+    config::SyncCommitteeSize,
 >;
```

Comment:
Actually, snowfork had two exact same set of configuration. One as constant and other configurable via pallet config. It was confusing as if you change the constants but do not change the pallet config the pallet won't work. 

Configuration pertaining to the size of the data structure inside the sync data needs to be a constant and we cannot use generic parameter inside the expression. The restriction comes from the library used to serialize and deserialize data structure to/from simple serialization format. (Unless we enable generic_const_exprs unstable feature, which is not ideal).

So, I just removed the configuration from the runtime and used constant parameter types to have same set of configuration. Yes, this restricts our ability to use the pallet, this is something I need to fix later on.

Reply:
'generic_const_exprs' is not an issue, we use a few unstable features already. If there was duplication, I'd imagine it to be de-duplicated in a way that extends functionality rather than hardcoding it and would upstream that to Snowfolk.

For instance, change constants to runtime configuration, that way all of those parameters could be configured in the chain spec, not requiring any features on pallet level and being as flexible as we want. But at least having runtime constants configurable from the outside is not as bad as just hardcoding them and making it impossible to have two runtimes in the workspace that require different feature flags.